src/modeler/AutoRest.Swagger/Validation/ListByOperationsValidation.cs

```diff
@@ -0,0 +1,163 @@
+                }
+
+                // find if there are any operations that violate the rule
+                var errOpIds = listOpIds.Where(opId => !opNames.Contains(opId.ToLower()));
+                
+                // no violations found, skip
+                if (IsNullOrEmpty(errOpIds))
+                {
+                    continue;
+                }
+                // aggregate suggested op names in a single readable string for the formatter
+                var suggestedNames = opNames.Aggregate((curr, next)=> curr+", "+ next);
+                foreach (var errOpId in errOpIds)
+                {
+                    var stringParams = new List<string>() { errOpId };
+                    stringParams.AddRange(opNames);
+                    yield return new ValidationMessage(new FileObjectPath(context.File, context.Path), this, errOpId, suggestedNames);
+                }
+            }
+        }
+
+        private List<string> GetValidListOpNames(string path)
+        {
+            var opList = new List<string>();
+            var methodGrp = GetMethodGroup(path);
+            if (methodGrp == string.Empty || !SubscriptionRegex.IsMatch(path))
+            {
+                return opList;
+            }
+
+            // construct _listbysubscriptionid
+            opList.Add(string.Format(ListByTemplate, methodGrp, SubscriptionId).ToLower());
+            opList.Add(string.Format(ListTemplate, methodGrp).ToLower());
+
+            // construct _listbyresourcegroup
+            if ((RgRegex.IsMatch(path)))
+            {
+                opList.Add(string.Format(ListByTemplate, methodGrp, ResourceGroup).ToLower());
+            }
+            else
+            {
+                return opList;
+            }
+            
+            // construct _listbyparent
+            var match = ParentRegex.Match(path.Substring(0, path.IndexOf(methodGrp)-1));
+            if (match.Success)
+            {
+                var caps = match.Groups["parent"].Captures.OfType<Capture>().Reverse();
+                var capture = caps.FirstOrDefault(cap => !PathParameterRegex.IsMatch(cap.Value));
+                // if due to some reason the parent we find is of the form "foo.bar", let's recommend the
+                // operation id be named as "_listbyfoobar"
+                if (capture != null)
+                {
+                    opList.Add(string.Format(ListByTemplate, methodGrp, capture.Value.Substring(1)).Replace(".", string.Empty).ToLower());
+                }
+            }
+            
+            return opList;
+        }
+
+        private string GetMethodGroup(string path)
+        {
+            var match = UrlRegex.Match(path);
+            var caps = match.Groups["urltoken"].Captures.OfType<Capture>().Reverse();
+            var capture = caps.FirstOrDefault(cap => !PathParameterRegex.IsMatch(cap.Value));
+            
+            return (capture!=null) ? capture.Value.Substring(1) : string.Empty;
```

Comment:
make this `capture?.Value.Substring(1)` and change the check `GetValidListOpNames` to check for `null`. Don't use an empty string as a "special" value